src/noding/snapround/SnapRoundingNoder.cpp

```diff
@@ -77,8 +77,9 @@ SnapRoundingNoder::snapRound(std::vector<SegmentString*>& inputSegStrings, std::
 void
 SnapRoundingNoder::addIntersectionPixels(std::vector<SegmentString*>& segStrings)
 {
-    SnapRoundingIntersectionAdder intAdder(pm);
-    MCIndexNoder noder;
+    double tolerance = 1.0 / pm->getScale() / INTERSECTION_NEARNESS_FACTOR;
+    SnapRoundingIntersectionAdder intAdder(tolerance);
+    MCIndexNoder noder(nullptr, tolerance);
     noder.setSegmentIntersector(&intAdder);
```

Comment:
You can merge these two lines, like is done [here](https://github.com/libgeos/geos/blob/main/src/noding/snap/SnappingNoder.cpp#L129).

Reply:
Done!  Can't believe I just ignored that nullptr like that.